### PR TITLE
Added versioning to E2E tests with new upgrade test template

### PR DESCRIPTION
**Pull request by Jani19966**
**Description**

Changes proposed in this pull request:

- Added a new test definition for the reason to upgrade a cluster from a specific version to another.
- Removed orchestration methods from e2e_upgrade as to another script does this job.
- Added base version to create the cluster at.

**Related issue(s)**

### CODE REVIEW

File: resources/kcp/charts/kyma-environment-broker/templates/tests/upg-tst-aws-prov.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  name: upg-tst-aws-prov
+  labels:
+{{ include "kyma-env-broker.labels" . | indent 4 }}
+spec:
+  disableConcurrency: false
+  template:
+    spec:
+      serviceAccountName: e2e-provisioning
+      volumes:
+        - name: gardener-kubeconfig
```
Comment (ebensom) [inline resources/kcp/charts/kyma-environment-broker/templates/tests/upg-tst-aws-prov.yaml:14]: Does the test case need the gardener kubeconfig?

File: resources/kcp/charts/kyma-environment-broker/templates/tests/upg-tst-aws-prov.yaml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+              value: "true"
+            - name: APP_BROKER_URL
+              value: 'https://{{ .Values.host }}.{{ .Values.global.ingress.domainName }}'
+            - name: APP_BROKER_PLAN_ID
+              value: "{{ .Values.e2e.aws.planID }}"
+            - name: APP_BROKER_CLIENT_NAME
+              value: {{ include "kyma-env-broker.fullname" . | quote }}
+            - name: APP_CONFIG_NAME
+              value: "upg-tst-aws-config"
+            - name: APP_PRE_UPGRADE_KYMA_VERSION
```
Comment (ebensom) [inline resources/kcp/charts/kyma-environment-broker/templates/tests/upg-tst-aws-prov.yaml:137]: Cleanup does not need this setting

**Review by ebensom — APPROVED** [review]

**Comment by ebensom** [comment]
You will also need to bump the e2e image version to PR-832 here:
https://github.com/kyma-project/control-plane/blob/main/resources/kcp/values.yaml#L34

**Review by ebensom — DISMISSED** [review]

**Review by ksputo — DISMISSED** [review]
lgtm

**Comment by ksputo** [comment]
Please bump the image here: https://github.com/kyma-project/control-plane/blob/3d09e705340ef9f454d7e437a9aae13f2bc65366/resources/kcp/values.yaml#L34 to `PR-832`

**Review by ksputo — APPROVED** [review]
lgtm

**Review by piotrmiskiewicz — APPROVED** [review]
lgtm

**Pull request merged**